chrome/content/zotero/integration/citationDialog/searchHandler.mjs

```diff
@@ -248,7 +248,18 @@ export class CitationDialogSearchHandler {
 			}
 			return Zotero.Cite.getItem(itemID);
 		});
-		return items;
+
+		// Deduplicate items since there may be multiple tabs opened per the same
+		// item (duplicate tabs or a PDF and a snapshot belonging to the same item)
+		let uniqueItems = [];
+		let seenItemIDs = new Set();
+		for (let item of items) {
+			if (!seenItemIDs.has(item.id)) {
+				uniqueItems.push(item);
+				seenItemIDs.add(item.id);
+			}
+		}
+		return uniqueItems;
```

Comment:
I would think `return [...new Set(items)];` should be sufficient.

Reply:
Oh, that's completely right! Because if there are two duplicate items in the array, it's just two pointers to the same object, as opposed to having two different objects. I re-pushed with this simpler approach.